airbyte-integrations/connectors/source-tplcentral/source_tplcentral/streams.py

```diff
@@ -0,0 +1,268 @@
+from urllib.parse import parse_qsl, urlparse
+
+import arrow
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+from source_tplcentral.util import deep_get, normalize
+
+
+class TplcentralStream(HttpStream, ABC):
+    url_base = None
+
+    def __init__(self, config) -> None:
+        super().__init__(authenticator=config["authenticator"])
+
+        self.url_base = config["url_base"]
+        self.customer_id = config.get("customer_id")
+        self.facility_id = config.get("facility_id")
+        self.start_date = config.get("start_date")
+
+        self.total_results_field = "TotalResults"
+
+    primary_key = "_id"
+
+    @property
+    def page_size(self):
+        None
+
+    @property
+    def upstream_primary_key(self):
+        None
+
+    @property
+    def upstream_cursor_field(self):
+        None
+
+    @property
+    @abstractmethod
+    def collection_field(self) -> str:
+        pass
+
+    def next_page_token(self, response: requests.Response, **kwargs) -> Optional[Mapping[str, Any]]:
+        data = response.json()
+        total = data[self.total_results_field]
+
+        pgsiz = self.page_size or len(data[self.collection_field])
+
+        url = urlparse(response.request.url)
+        qs = dict(parse_qsl(url.query))
+
+        pgsiz = int(qs.get("pgsiz", pgsiz))
+        pgnum = int(qs.get("pgnum", 1))
+
+        if pgsiz > 0 and pgsiz * pgnum < total:
+            return {
+                "pgsiz": pgsiz,
+                "pgnum": pgnum + 1,
+            }
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return next_page_token
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        records = normalize(response.json()[self.collection_field])
+
+        for record in records:
+            if self.upstream_primary_key:
+                record[self.primary_key] = deep_get(record, self.upstream_primary_key)
+            if self.upstream_cursor_field:
+                record[self.cursor_field] = deep_get(record, self.upstream_cursor_field)
+            yield record
+
+
+class StockSummaries(TplcentralStream):
+    collection_field = "Summaries"
+    primary_key = ["facility_id", "_item_identifier_id"]
+    page_size = 500
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "inventory/stocksummaries"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        records = super().parse_response(response, **kwargs)
+
+        for record in records:
+            record["_item_identifier_id"] = deep_get(record, "ItemIdentifier.Id")
+            yield record
+
+
+class Customers(TplcentralStream):
```

Comment:
can you add comments about the URL for documentation to this stream. Same for all the streams